kratos/conditions/geometrical_condition.cpp

```diff
@@ -0,0 +1,146 @@
+/***********************************************************************************/
+
+GeometricalCondition::GeometricalCondition(
+    IndexType NewId, 
+    GeometryType::Pointer pGeometry
+    ) : BaseType(NewId, pGeometry)
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+GeometricalCondition::GeometricalCondition(
+    IndexType NewId, 
+    GeometryType::Pointer pGeometry, 
+    PropertiesType::Pointer pProperties
+    ) : BaseType(NewId,pGeometry, pProperties)
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+GeometricalCondition::GeometricalCondition(GeometricalCondition const& rOther)
+    : BaseType(rOther)
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+GeometricalCondition::~GeometricalCondition()
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+GeometricalCondition& GeometricalCondition::operator=(GeometricalCondition const& rOther)
+{
+    //ALL MEMBER VARIABLES THAT MUST BE KEPT IN AN "=" OPERATION NEEDS TO BE COPIED HERE
+
+    Condition::operator=(rOther);
+
+    return *this;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Condition::Pointer GeometricalCondition::Create(
+    IndexType NewId, 
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties
+    ) const
+{
+    KRATOS_TRY
+    return Kratos::make_shared<GeometricalCondition>(NewId, GetGeometry().Create(ThisNodes), pProperties);
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Condition::Pointer GeometricalCondition::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    PropertiesType::Pointer pProperties
+    ) const
+{
+    KRATOS_TRY
+    return Kratos::make_shared<GeometricalCondition>(NewId, pGeom, pProperties);
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Condition::Pointer GeometricalCondition::Clone (
+    IndexType NewId, 
+    NodesArrayType const& ThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    Condition::Pointer p_new_elem = Kratos::make_shared<Condition>(NewId, GetGeometry().Create(ThisNodes), pGetProperties()); 
```

Comment:
This is creating a base class condition! Kratos::make_shared<GeometricalCondition>
Also, maybe p_new_condition would be a more descriptive name...

Reply:
Copy paste error, sorry